### DISCUSSION TITLE
We did an accessibility audit on Radix UI and want to know what the best next step is.

### DISCUSSION THREAD

**Discussion by zoltan-dulac** [radix-ui/primitives#2232] [Ideas]
Good Afternoon.

I am a member of Publicis Sapient's Accessibility Center of Excellence, and since our front-end team is interested in using your library on client work, we decided to do an audit of it, testing the component library using a number of screen reader/browser pairs:

- Voicover/Safari on OSX and iOS
- NVDA/Firefox on Windows.
- Talkback/Chrome on Android.

We have a list of 35 accessibility issues that we would like to open.  These are currently in a spreadsheet, but before we do the laborious job of cutting and pasting them into the Issues tab on this project's GitHub, we thought we'd better ask if this is the best way to give feedback, or if you'd like to review the spreadsheet first.

Please note that we may be able to fix these ourselves, but we have to allocate time and figure out budget for this.  Before we do that, what say you on what the best next steps should be?

Please let us know: we are really eager to help if possible.  Thanks for your consideration.

Zoltan.

**Comment by cusserycms** [comment]
Have the same questions as above

**Comment by BleedingDev** [comment]
Was there any outcome? :)

**Comment by lindelleric** [comment]
Any news on this?

**Comment by Casperjuel** [comment]
[zoltan-dulac](https://github.com/zoltan-dulac) did you guys move forward with any of this?

**Comment by zoltan-dulac** [comment]
@Casperjuel As I mentioned to @BleedingDev in an email today: We still have the spreadsheet mentioned in this issue's description. After doing a super-fast review over the tabs in the spreadsheet, it looks like the issues are still there in 2025.  Here is the spreadsheet:
[Radix-ui_accessibility-review.xlsx](https://github.com/user-attachments/files/18914088/Radix-ui_accessibility-review.xlsx)

Each tab inside the spreadsheet represents a component.  In each tab, you'll see:

1. A list of screen reader/browser pairs that we tested with.
2. A list of defects for the component, with the WCAG guideline violated, the defect description and priority of fixing.
3. The priority definitions are as follows:

P1 (Critical): A barrier exists with result that some content or functionality cannot be accessed by a user with disabilities
P2 (Serious): A barrier exists with result that some content or functionality requires a workaround to be accessed for user  with disabilities. 
P3 (Moderate): Features are missing that would make the experience easier to use for a user with disabilities. (e.g. Lack of consistency in visible focus indicator). Minor issues that occur at a global level, or more serious issues (P1 or P2) that only occur in edge cases (e.g. for a single breakpoint, or on a single page). 
P4 (Minor): Minor issues that damage credibility but are unlikely to cause problems. (E.g. Redundant alternate text).

I'm sure there are going to be questions, which we can definitely discuss.  Please feel free to drop me a note here or to zoltan.dulac@gmail.com



/cc @BleedingDev @lindelleric @cusserycms

**Reply by Casperjuel** [reply]
Thanks for the update. Let’s hope the maintainers (@chaance)? prioritize this very soon. ❤️

**Reply by chaance** [reply]
@zoltan-dulac Amazing work, really appreciate this! Sorry no one got back to you sooner. We'll take a look at these issues and handle prioritizing from here. PRs are always welcome in the mean time!

**Comment by unjust** [comment]
Any outcome or action items on this?
